### PR TITLE
Allow using --watch option in patternlab build

### PR DESCRIPTION
**Pull request by drhossa**
Summary of changes:

- core/src/lib/patternlab.js: reset patterns before updating
- core/src/lib/watchPatternLabFiles.js: add unlink events to patternWatcher
- cli/bin/build.js: allow passing the watch flag

### CODE REVIEW

**Comment by bmuenzenmeyer** [comment]
this looks awesome! hope to review soon

File: packages/core/src/lib/patternlab.js [hunk 1]
```diff
@@ -307,6 +307,11 @@ module.exports = class PatternLab {
   // dive once to perform iterative populating of patternlab object
   processAllPatternsIterative(patterns_dir) {
     const self = this;
+    
+    // before updating the patterns has to be reset, otherwise
+    // deleted pattern would still be present in the patterns array
+    this.patterns = [];
+
```
Comment (JosefBredereck) [inline packages/core/src/lib/patternlab.js]: How does this work with the incremental build, when `cleanPublic: false` is set?

**Comment by mfranzke** [comment]
Fight the bot

**Comment by JosefBredereck** [comment]
Need to mention
Closes #887

**Pull request merged**